File: AssetInsight/src/main/java/br/unipar/assetinsight/dtos/requests/ServicoRequest.java
package br.unipar.assetinsight.dtos.requests;
import br.unipar.assetinsight.entities.OrdemServicoEntity;
import br.unipar.assetinsight.entities.ServicoEntity;

import java.io.Serializable;

/**
 * DTO para requests post de {@link ServicoEntity} em {@link OrdemServicoEntity}
 */
public record ServicoRequest(
        Long id,
        String patrimonio,
        String descricao,
        long quantidade,
        double valorUnit,
        double valorTotal,
        Long categoria,
        Long ambiente
) implements Serializable { }

File: Frontend/src/app/dtos/requests/servico.request.ts
export class ServicoRequest {
  id?: number | any;
  patrimonio?: number | any;
  descricao?: string | any;
  quantidade?: number | any;
  valorUnit?: number | any;
  categoria?: number | any;
  ambiente?: number | any;

  constructor() {}

  setValues(
    id?: number | any,
    patrimonio?: number | any,
    descricao?: string | any,
    quantidade?: number | any,
    valorUnit?: number | any,
    categoria?: number | any,
    ambiente?: number | any
  ) {
    this.id = id;
    this.patrimonio = patrimonio;
    this.descricao = descricao;
    this.quantidade = quantidade;
    this.valorUnit = valorUnit;
    this.categoria = categoria;
    this.ambiente = ambiente;
  }
}